### PR TITLE
Add options to fetchById and getOrFetch

### PR DESCRIPTION
**Pull request by dhritzkiv**
Plus minor cleanup

### CODE REVIEW

**Comment by dhritzkiv** [comment]
(Tests are broken as it relies on #37 to fix bind)

**Comment by wraithgar** [comment]
It looks like currently `options.success` and `options.error` are things that only `fetch` preserves.  They would have been passed in by coincidence on the old `fetchById` and clobbered if `options.all` were passed to `getOrFetch`.

Now, they are clobbered in `fetchById` and preserved by coincidence in `getOrFetch`.

Perhaps we could be a bit more explicit about where we preserve those, i.e. having `fetchById` mimic `fetch` in preserving any `options.success` or `options.error` passed in?

**Comment by dhritzkiv** [comment]
I agree with that. Check out my latest commit. Also, I made some (non-breaking) changes, including some that weren't necessary, but ones that helped me with readability and understanding through consistency. Happy to revert parts of it if needed.

File: ampersand-collection-rest-mixin.js [hunk 1]
```diff
@@ -18,15 +9,23 @@ module.exports = {
...
+            if (options.set !== false) self.trigger('sync', self, resp, options);
+        };
+        
+        // Wrap an optional error callback with a fallback error event.
+        var error = options.error;
+        options.error = function(resp) {
+            if (error) error(self, resp, options);
+            self.trigger('error', self, resp, options);
         };
-        wrapError(this, options);
```
Comment (wraithgar) [inline ampersand-collection-rest-mixin.js]: In the past this wrapError has proven to be one abstraction over the line of what I was able to follow, and wholeheartedly agree w/ the direction taken here of just inlining it where needed.

**Comment by wraithgar** [comment]
Yeah that looks good now, very easy to follow and preserves success/error options if passed in.

+1 from me. Seems like a minor release which reminds me that the readme should also show options as a valid param for fetchById.

**Comment by dhritzkiv** [comment]
I will make changes to the readme to reflect this

**Comment by wraithgar** [comment]
Great, that commit should re-kick tests which hopefully pass now that #37 is merged

**Comment by dhritzkiv** [comment]
I've added the `options` argument to fetchById. I've also mentioned the `options.all` property for `getOrFetch`, as well as the availability of `always` from ampersand-sync

**Comment by kamilogorek** [comment]
Looks fine!

**Comment by wraithgar** [comment]
published as `v5.2.0`

**Pull request merged**